unittest/unit/update-bcd.ts

```diff
@@ -861,5 +861,205 @@ describe('BCD updater', () => {
         version_added: null
       });
     });
+
+    describe('mirror', () => {
+      const browsers: any = {
+        chrome: {name: 'Chrome', releases: {85: {}, 86: {}}},
+        chrome_android: {
+          name: 'Chrome Android',
+          upstream: 'chrome',
+          releases: {86: {}}
+        }
+      };
+
+      const bcdFromSupport = (support) => ({
+        api: {FakeInterface: {__compat: {support}}}
+      });
+
+      const mirroringCase = ({support, downstreamResult}) => {
+        const reports: Report[] = [
+          {
+            __version: '0.3.1',
+            results: {
+              'https://mdn-bcd-collector.appspot.com/tests/': [
+                {
+                  name: 'api.FakeInterface',
+                  exposure: 'Window',
+                  result: downstreamResult
+                }
+              ]
+            },
+            userAgent:
+              'Mozilla/5.0 (Linux; Android 10; SM-G960U) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.5112.97 Mobile Safari/537.36'
+          }
+        ];
+        const supportMatrix = getSupportMatrix(reports, browsers, []);
+        const bcd = bcdFromSupport(support);
+        update(bcd, supportMatrix, {});
+        return bcd;
+      };
+
+      describe('supported upstream (without flags)', () => {
+        it('supported in downstream test results', () => {
+          const actual = mirroringCase({
+            support: {
+              chrome: {version_added: '85'},
+              chrome_android: 'mirror'
+            },
+            downstreamResult: true
+          });
+          assert.deepEqual(
+            actual,
+            bcdFromSupport({
+              chrome: {version_added: '85'},
+              chrome_android: {version_added: '86'}
```

Comment:
Without results from Chrome Android 85, I don't think this is a valid assumption. The updater script should have concluded "support in Chrome Android 86, earlier support unknown" and not see a contradiction in need of fixing by updating the data.

The updater code isn't really structured in a way that makes this easy to fix however, see https://github.com/foolip/mdn-bcd-collector/issues/571

Reply:
> Without results from Chrome Android 85, I don't think this is a valid assumption. The updater script should have concluded "support in Chrome Android 86, earlier support unknown" and not see a contradiction in need of fixing by updating the data.

I think I understand. The expected value for this test should be "mirror", right?

> The updater code isn't really structured in a way that makes this easy to fix however, see #571

The code currently produces the value "mirror" for this case (as per the first commit on this branch), so if that matches the expectation, then there's nothing to fix here.

Reply:
Yes, the way I'd like the updater to work is that when we have incomplete information that doesn't contradict the existing data, it is left alone. (An early goal of being able to derive everything in BCD from test results has gone out the window by now, but might explain some choices in the design.)

What you might want to do is to tweak these tests to actually work with a false-to-true transition, to test the behavior I think you were originally going for here.

The case of incomplete information can be tested much more lightly, maybe just one "no change made" test, and perhaps one for the case  where we can say the current data is wrong, but not exactly what is right. Such tests need not involve mirroring though.

Reply:
How to deal with incomplete information is indeed beyond the scope of fixing mirroring, but the test expectations you've written seem to suggest you'd like to change it, unless I'm wrong about current behavior.

The way I'd suggest approaching mirroring is:
- Mirror the data from upstream
- Do everything the same way as if mirroring didn't exist
- Optionally at the end, detect if the new data is the same as mirroring would produce and if so replace with "mirror". (This isn't super important because one can always run the mirror migration script in BCD later to fix it.)

My comments are about how the "do everything" bit should work, since the tests poke at that. The "transition" I'm talking about is not changing a value in BCD, but the logic in `inferSupportStatements`. If we have results for Chrome Android 85 and 86 and observe a false&rarr;true transition in test results between the two, then we can infer `{version_added: "86"}`. But if we have results from *only* 86, then the correct inference is `{version_added: "≤86"}`. That wont pass the BCD lint, but it does force a human to look into the uncertainty.

If you generate results for both 85 and 86 you can deal only with the simpler case, which I suspect is the more important one to get fixed.

Reply:
My goal is to contribute data about the Meta Quest browser (formerly Oculus Quest). I'm focused on the mirroring behavior of `update-bcd` script because it's my understanding that (due to [this patch](https://github.com/mdn/browser-compat-data/pull/16895)), I'll be relying heavily on that specific functionality. As I understand it, the incomplete information in this case would be compatibility data for prior releases of the Meta Quest browser.

That's why all of these tests describe conditions where the "downstream" support value is set to `"mirror"` and why none include prior support values downstream. I thought I was covering all the conditions for my use case exhaustively, but I'm getting the sense that either my understanding of the use case is inaccurate or that the underlying problem is larger than what can be observed (or addressed) by my use case alone.

Does that match your understanding?

Reply:
I think I might be getting you entangled in some details that I enjoy but that aren't super relevant to you, like what exactly to do with partial test results.

Perhaps the concrete bug you're seeing would help guide us to the right things to test. I guess you're running `update-bcd` with some results from Meta Quest, and it's making the wrong change in some cases? Can we start with one case to get warmed up?

Reply:
I can't collect test results from that browser at the moment; my participation in this repository has so far been proactive. I've been trying to enumerate all of the situations that the script might encounter when it receives the first set of results from the Quest browser. I may have identified some conditions which are more exotic than we'll see with authentic data, but I'm fairly confident that many represent mainline use cases. Any problems with those will block progress when I do have results in hand.

An example would be when a test result for a "downstream" browser is negative, and the "upstream" support is positive. The script ought to change `"mirror"` to `{version_added: false}`, but it currently does nothing. (That case is described by the second row of the table in gh-2261 and by the test named "BCD updater, update, mirror, supported upstream (without flags), unsupported in downstream test results" in this patch.)

It may be that we're closer than we think. Between [implementing support for the new "flags" data policy](https://github.com/foolip/mdn-bcd-collector/pull/2297), learning that existing support data should be ignored, and generally understanding the internals a bit better, I've been able to write a patch that corrects the problems. I'll be able to share it here once the "flags" patch is merged. In the mean time, [there's one more expected value I'd like to revisit](https://github.com/foolip/mdn-bcd-collector/pull/2280/commits/5f3e7f3ed6e7f299e18b04f946dd27be07a8bed7#r961206168).

Reply:
Yes, we should change `"mirror"` to `{version_added: false}` if applying the mirroring results in enabled-by-default support, which is contradicted by test results. This is the framework I'd suggest in general, to adjust data to be consistent with the evidence, and otherwise leave the data alone, even if we're not able to verify it.